src/test/java/de/unistuttgart/chickenshockbackend/ConfigControllerTest.java

```diff
@@ -311,4 +311,31 @@ void updateQuestionFromExistingConfigurationNotFound() throws Exception {
             .andExpect(status().isNotFound())
             .andReturn();
     }
+
+    @Test
+    void testCloneConfiguration() throws Exception {
+        final MvcResult result = mvc
+            .perform(
+                post(API_URL + "/" + initialConfig.getId() + "/clone")
+                    .cookie(cookie)
+                    .contentType(MediaType.APPLICATION_JSON)
+            )
+            .andExpect(status().isCreated())
+            .andReturn();
+        final String content = result.getResponse().getContentAsString();
+        final UUID clonedId = objectMapper.readValue(content, UUID.class);
+        assertNotEquals(initialConfig.getId(), clonedId);
+
+        final Configuration cloneConfig = configurationRepository.findById(clonedId).get();
+        cloneConfig
+            .getQuestions()
+            .forEach(question ->
+                initialConfig
+                    .getQuestions()
+                    .forEach(initialQuestion -> {
+                        assertNotEquals(question.getId(), initialQuestion.getId());
+                    })
+            );
+        assertNotEquals(cloneConfig, initialConfig);
```

Comment:
Erm… This test looks somewhat irrelevant in this implementation.
You should also check that all questions are still present, and that the db also knows about this new configuration.

Reply:
This test, if the cloned questions are a deep copy. I think its important to test especially if the implementation gets changed.